### PR TITLE
Add The Things Uno

### PR DESCRIPTION
**Pull request by benolayinka**
#### Summary

Add The Things Uno

#### Release Notes

- Add The Things Uno

### CODE REVIEW

File: vendor/the-things-products/the-things-uno-codec.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
+uplinkDecoder:
+  fileName: the-things-uno.js
```
Comment (johanstokking) [inline vendor/the-things-products/the-things-uno-codec.yaml:2]: The problem with this device is that people can customize the firmware. There are quite some (examples)[https://github.com/TheThingsNetwork/arduino-device-lib/tree/master/examples] that all have different payload codecs.

I don't think we can declare a codec for The Things Uno.

File: vendor/the-things-products/the-things-uno.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  - version: '1.0'
+firmwareVersions:
+  - version: '1.0'
+    hardwareVersions:
+      - '1.0'
+    profiles:
+      EU863-870:
+        id: the-things-uno-eu868
+        lorawanCertified: true
+        codec: the-things-uno-codec
```
Comment (johanstokking) [inline vendor/the-things-products/the-things-uno.yaml:13]: So remove this

File: vendor/the-things-products/the-things-uno.yaml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+      US902-928:
+        id: the-things-uno-us915
+        lorawanCertified: true
+        codec: the-things-uno-codec
```
Comment (johanstokking) [inline vendor/the-things-products/the-things-uno.yaml:17]: Idem, and the codec definition file and script

**Review by johanstokking — CHANGES_REQUESTED** [review]

File: vendor/the-things-products/the-things-uno-codec.yaml (hunk 1 above)
Comment (benolayinka) [reply vendor/the-things-products/the-things-uno-codec.yaml:2]: can we declare that it matches the Quick Start arduino example?

Comment (johanstokking) [reply vendor/the-things-products/the-things-uno-codec.yaml:2]: Yes, but then I would declare that as a firmware version. So different sketches get a different firmware version; sketches are basically just that.

Comment (johanstokking) [inline vendor/the-things-products/the-things-uno-codec.yaml:2]: This should now be `the-things-uno-codec-quickstart.yaml`

File: vendor/the-things-products/the-things-uno.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
+// This corresponds to the Quick Start sketch in The Things Network arduino library
```
Comment (johanstokking) [inline vendor/the-things-products/the-things-uno.js:1]: This should now be `the-things-uno-quickstart.js`

Comment (johanstokking) [reply vendor/the-things-products/the-things-uno.js:1]: Maybe add a direct link to the QuickStart sketch on GitHub here

**Review by johanstokking — COMMENTED** [review]
Getting there!

**Review by johanstokking — APPROVED** [review]
Fixup and merge!

**Comment by benolayinka** [comment]
> Fixup and merge!

Only you got the power

Screenshot 2020-11-18 at 13 47 37

**Comment by johanstokking** [comment]
We fixed validation and this one doesn't pass, please fix

**Comment by benolayinka** [comment]
> We fixed validation and this one doesn't pass, please fix

Fixed

**Review by johanstokking — APPROVED** [review]
Awesome stuff

**Pull request merged**